### PR TITLE
Change from Lodash to Immutable

### PR DESCRIPTION
**Pull request by rrees**
Inspired by the wave of recent changes from @regiskuckaertz I thought I'd try and clean up some of the remaining dependencies on Lodash and also constant declarations that we make that are not explicitly Immutable.

### CODE REVIEW

File: src/element.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
-define(['lodash-amd/modern/collection/contains'], function (contains) {
+define(['immutable/dist/immutable'], function (Immutable) {
```
Comment (cutandpastey) [inline src/element.js]: Can we please reference in the require config this like the other dependencies?

**Comment by regiskuckaertz** [comment]
Awesome :+1: I have a rewrite of UndoManager with Immutable, let me know if it can help

**Comment by cutandpastey** [comment]
Have spoken to @rrees, we will alias Immutable in a separate PR. :+1:

File: src/element.js (hunk 1 above)
Comment (theefer) [inline src/element.js]: Good call

**Pull request merged**